modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/provider/CompareVersionsJournalArticleContentDashboardItemVersionActionProvider.java

```diff
@@ -0,0 +1,111 @@
+ * details.
+ */
+
+package com.liferay.content.dashboard.journal.internal.item.action.provider;
+
+import com.liferay.content.dashboard.item.action.ContentDashboardItemVersionAction;
+import com.liferay.content.dashboard.item.action.provider.ContentDashboardItemVersionActionProvider;
+import com.liferay.content.dashboard.journal.internal.item.action.CompareVersionsJournalArticleContentDashboardItemVersionAction;
+import com.liferay.journal.model.JournalArticle;
+import com.liferay.journal.service.JournalArticleLocalService;
+import com.liferay.journal.service.JournalArticleService;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Jürgen Kappler
+ */
+@Component(service = ContentDashboardItemVersionActionProvider.class)
+public class
+	CompareVersionsJournalArticleContentDashboardItemVersionActionProvider
+		implements ContentDashboardItemVersionActionProvider<JournalArticle> {
+
+	@Override
+	public ContentDashboardItemVersionAction
+		getContentDashboardItemVersionAction(
+			JournalArticle journalArticle,
+			HttpServletRequest httpServletRequest) {
+
+		if (!isShow(journalArticle, httpServletRequest)) {
+			return null;
+		}
+
+		return new CompareVersionsJournalArticleContentDashboardItemVersionAction(
+			httpServletRequest, journalArticle,
+			_getLatestApprovedJournalArticle(journalArticle), _language);
+	}
+
+	@Override
+	public boolean isShow(
+		JournalArticle journalArticle, HttpServletRequest httpServletRequest) {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		PermissionChecker permissionChecker =
+			themeDisplay.getPermissionChecker();
+
+		User user = themeDisplay.getUser();
+
+		int countArticles = _journalArticleService.getArticlesCountByArticleId(
+			journalArticle.getGroupId(), journalArticle.getArticleId());
+
+		JournalArticle latestJournalArticle = _getLatestApprovedJournalArticle(
+			journalArticle);
+
+		if (permissionChecker.isContentReviewer(
+				user.getCompanyId(), themeDisplay.getScopeGroupId()) &&
+			(countArticles > 1) &&
+			(journalArticle.getVersion() !=
+				latestJournalArticle.getVersion())) {
+
+			return true;
+		}
+
+		return false;
+	}
+
+	private JournalArticle _getLatestApprovedJournalArticle(
+		JournalArticle journalArticle) {
+
+		return _journalArticleLocalService.fetchLatestArticle(
+			journalArticle.getResourcePrimKey(), WorkflowConstants.STATUS_ANY);
+	}
+
+	@Reference
```

Comment:
@Stephy6, is there a way to use one service? Either JournalArticleLocalService or JournalArticleService. If not, then it's okay